samza-core/src/main/java/org/apache/samza/container/LocalityManager.java

```diff
@@ -35,56 +33,25 @@
  * Locality Manager is used to persist and read the container-to-host
  * assignment information from the coordinator stream
  * */
-public class LocalityManager extends AbstractCoordinatorStreamManager {
+public class LocalityManager {
   private static final Logger log = LoggerFactory.getLogger(LocalityManager.class);
   private Map<String, Map<String, String>> containerToHostMapping = new HashMap<>();
   private final TaskAssignmentManager taskAssignmentManager;
   private final boolean writeOnly;
+  private CoordinatorStreamManager coordinatorStreamManager;
+  private static final String CONTAINER_PREFIX = "SamzaContainer-";
 
-  /**
-   * Default constructor that creates a read-write manager
-   *
-   * @param coordinatorStreamProducer producer to the coordinator stream
-   * @param coordinatorStreamConsumer consumer for the coordinator stream
-   */
-  public LocalityManager(CoordinatorStreamSystemProducer coordinatorStreamProducer,
-                         CoordinatorStreamSystemConsumer coordinatorStreamConsumer) {
-    super(coordinatorStreamProducer, coordinatorStreamConsumer, "SamzaContainer-");
-    this.writeOnly = coordinatorStreamConsumer == null;
-    this.taskAssignmentManager = new TaskAssignmentManager(coordinatorStreamProducer, coordinatorStreamConsumer);
-  }
-
-
-  /**
-   * Special constructor that creates a write-only {@link LocalityManager} that only writes
-   * to coordinator stream in {@link SamzaContainer}
-   *
-   * @param coordinatorStreamSystemProducer producer to the coordinator stream
-   */
-  public LocalityManager(CoordinatorStreamSystemProducer coordinatorStreamSystemProducer) {
-    this(coordinatorStreamSystemProducer, null);
-  }
 
   /**
-   * This method is not supported in {@link LocalityManager}. Use {@link LocalityManager#register(String)} instead.
+   * Constructor that creates a read-write or write-only locality manager.
    *
-   * @throws UnsupportedOperationException in the case if a {@link TaskName} is passed
+   * @param coordinatorStreamManager Coordinator stream manager.
+   * @param writeOnly True if write-only, otherwise false.
    */
-  @Override
-  public void register(TaskName taskName) {
-    throw new UnsupportedOperationException("TaskName cannot be registered with LocalityManager");
-  }
-
-  /**
-   * Registers the locality manager with a source suffix that is container id
-   *
-   * @param sourceSuffix the source suffix which is a container id
-   */
-  public void register(String sourceSuffix) {
-    if (!this.writeOnly) {
-      registerCoordinatorStreamConsumer();
-    }
-    registerCoordinatorStreamProducer(getSource() + sourceSuffix);
+  public LocalityManager(CoordinatorStreamManager coordinatorStreamManager, boolean writeOnly) {
+    this.coordinatorStreamManager = coordinatorStreamManager;
```

Comment:
Does this need to register itself with the CoordinatorStreamManager as well?

Reply:
I think I misunderstood, looks like other Managers don't need to register _themselves_ with the CoordinatorStreamManager. We just need to call register on the CoordinatorStreamManager. In that case this makes sense. 

Although, in that case, why does the register on CSM exist? Couldn't its job be done in the constructor?

Reply:
Most cases, yes, but if you look at `SamzaContainer`, the source name to register with is not known until the container ID is known from the `containerContext`. I'm sure all of that can be refactored. Also one more thing is that register does a fetch on the consumer, if there is a consumer, so I'm not sure doing IO in the constructor is a good idea.